docs/backend-requests/handling/fastify.mdx

```diff
@@ -0,0 +1,175 @@
+    const auth = getAuth(request);
+    if (!auth.userId) {
+      return reply.code(401).send(null);
+    }
+    return reply;
+  };
+
+  const privateRoutes = async (fastify) => {
+    fastify.addHook('preHandler', clerkRequireAuth);
+
+    fastify.get('/private', async (req, _reply) => {
+      return { hello: 'world', auth: getAuth(req) };
+    });
+  };
+
+  server.register(privateRoutes);
+
+  const start = async () => {
+    try {
+      await server.listen({ port: 3000 });
+    } catch (err) {
+      server.log.error(err);
+      process.exit(1);
+    }
+  };
+  start();
+  ```
+
+  ```typescript
+  import 'dotenv/config'; // To read CLERK_PUBLISHABLE_KEY & CLERK_SECRET_KEY
+  import Fastify from 'fastify';
+  import { clerkPlugin, getAuth } from '@clerk/fastify';
+  import type { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+
+  const server = Fastify({ logger: true });
+
+  server.register(clerkPlugin);
+
+  const clerkRequireAuth = (request: FastifyRequest, reply: FastifyReply) => {
+    const auth = getAuth(request);
+    if (!auth.userId) {
+      return reply.code(401).send(null);
+    }
+    return reply;
+  };
+
+  const privateRoutes = async (fastify: FastifyInstance) => {
+    fastify.addHook('preHandler', clerkRequireAuth);
+
+    fastify.get('/private', async (req: FastifyRequest, _reply: FastifyReply) => {
+      return { hello: 'world', auth: getAuth(req) };
+    });
+  };
+
+  server.register(privateRoutes);
+
+  const start = async () => {
+    try {
+      await server.listen({ port: 3000 });
+    } catch (err) {
+      server.log.error(err);
+      process.exit(1);
+    }
+  };
+  start();
+  ```
+</CodeBlockTabs>
+  
+
+## Plugin options
+
+These options can be used `clerkPlugin`.
+
+| Name        | Type        | Description    |
+| ----------- | ----------- | -------------- |
+| `jwtKey`  | `string` | Clerk's JWT session token can be verified in a networkless manner using the JWT verification key. By default, Clerk will use our well-known JWKs endpoint to fetch and cache the key for any subsequent token verification. If you use the `CLERK_JWT_KEY` environment variable or the `jwtKey` option to supply the key, Clerk will pick it up and do networkless verification for session tokens using it. For more information, refer to [Networkless Token Verification](https://clerk.com/docs/references/nodejs/token-verification#validate-the-authorized-party-of-a-session-token). |
+| `hookName` | `string` | Can be `onRequest` or `preHandler`  |
+| `secretKey` | `string` | TODO  |
+| `apiUrl` | `string` | TODO  |
+| `apiVersion` | `string` | TODO  |
+| `userAgent` | `string` | TODO  |
+| `audience` | `string \| string[]` | TODO  |
+| `publishableKey` | `string` | TODO  |
+| `isSatellite` | `boolean` | TODO  |
+| `domain` | `string` | TODO  |
+| `proxyUrl` | `string` | TODO  |
+| `sdkMetadata` | `{ name: string, version: string }` | TODO  |
+| `telemetry` | `{ disabled: boolean, debug: boolean }` | TODO  |
```

Comment:
if this PR isn't ready for review, please revert to draft 🫶
also, I'm pretty sure we have the descriptions for these values on other pages! I'd ctrl + shift + f to search all of the clerk docs and just copy and paste the descriptions :)

Reply:
i have reverted it to draft since it's not ready for review 😓 
Yes, i think that I've seen most of them somewhere in our docs.